components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,464 @@
+			placement = placement + 1
+			lastScore = opponent.score
+			lastPlacement = opponent.placement
+			lastStatus = opponent.status
+		end
+	-- luacheck: pop
+	-- only apply arg changes otherwise
+	else
+		for opponentIndex, opponent in pairs(opponents) do
+			args['opponent' .. opponentIndex] = opponent
+		end
+	end
+	if
+		winner == RESULT_TYPE_DRAW or
+		winner == '0' or (
+			Logic.readBool(args.finished) and
+			#opponents == 2 and
+			opponents[1].status == STATUS_HAS_SCORE and
+			opponents[2].status == STATUS_HAS_SCORE and
+			opponents[1].score == opponents[2].score
+		)
+	then
+		args.winner = 0
+		args.resulttype = RESULT_TYPE_DRAW
+	elseif
+		Logic.readBool(args.finished) and
+		#opponents == MAX_NUM_OPPONENTS and
+		opponents[1].status ~= STATUS_HAS_SCORE and
+		opponents[1].status == opponents[2].status
+	then
+		args.winner = 0
+	end
+	return args
+end
+
+function matchFunctions.getPlayers(match, opponentIndex, teamName)
+	for playerIndex = 1, MAX_NUM_PLAYERS do
+		-- parse player
+		local player = Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex]) or {}
+		player.name = player.name or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.flag = player.flag or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'flag')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. playerIndex] = player
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+function mapFunctions.getExtraData(map)
+	local overtimes = Array.extractValues(Table.mapValues(mw.text.split(map.overtime or '', ','), tonumber))
+
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+		overtime = Table.isNotEmpty(overtimes) and Json.stringify(overtimes) or nil
+	}
+	return map
+end
+
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				score = tonumber(score)
+				obj.status = STATUS_HAS_SCORE
+				obj.score = score
+				obj.index = scoreIndex
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+				obj.index = scoreIndex
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+	local isFinished = map.finished
+	if not Logic.isEmpty(isFinished) then
+		isFinished = Logic.readBool(isFinished)
+	else
+		isFinished = not Logic.readBool(map.unfinished)
+	end
+	if isFinished and not Logic.isEmpty(indexedScores) then
+		map.winner = mapFunctions.getWinner(indexedScores)
+	end
+
+	return map
+end
+
+function mapFunctions.getWinner(indexedScores)
+	table.sort(indexedScores, mapFunctions.mapWinnerSortFunction)
+	return indexedScores[1].index
+end
+
+function mapFunctions.mapWinnerSortFunction(op1, op2)
```

Comment:
This is basically the same as `CustomMatchGroupInput._placementSortFunction`, look at merging the two, or at least the duplicated parts of them.

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/26bce035ec6c9d8cd6d19e1776107c9ec9b7b01a